fdbserver/fdbserver.actor.cpp

```diff
@@ -1599,6 +1599,11 @@ int main(int argc, char* argv[]) {
 			flushAndExit(FDB_EXIT_ERROR);
 		}
 
+#if defined(__linux__) // is there a better way to detect glibc malloc?
+		// need to do this before we start spawning threads, so before we start running flow actors.
+		mallopt(M_ARENA_MAX, SERVER_KNOBS->SERVER_MALLOC_ARENA_MAX);
```

Comment:
Am I allowed to emit a trace event this early in execution?

mallopt can return an error (basically just "i failed"), and I'd like to log it as a warning.

Reply:
I think so, because there are trace events above this point in `main()` and there are trace events in `CLIOptions::parse()` which is called above this point.

Reply:
I've been trying to decide if this is the right place for this option to be set and what the default value should be.

Normally, FDB creates very few threads and most work is done on the main thread.  The TLS handshake threads are only created when TLS is initialized, which happens on-demand upon the first creation of a TLS listener or TLS connection to a remote peer.  So one could argue that the arena limit should only be set if there are more TLS threads configured than the arena max and that it should be set during TLS initialization.

If, for example, one were using a storage engine that uses many background threads (such as the new RocksDB adapter) and were either not using TLS, then reducing the arena count to 6 would probably hurt performance.

Reply:
I'm thinking this option should be tied to the TLS options, only set during TLS initialization and only if the TLS handshake thread count is greater than zero.